zstd-safe/zstd-sys/build.rs

```diff
@@ -114,6 +118,14 @@ fn compile_zstd() {
         config.file("zstd/lib/decompress/huf_decompress_amd64.S");
     }
 
+    if env::var("CARGO_CFG_TARGET_ARCH").ok() == Some("wasm32".into()) {
```

Comment:
Since we're re-checking this further down, I would save it as a variable.

```suggestion
    let is_wasm32 = env::var("CARGO_CFG_TARGET_ARCH").ok() == Some("wasm32".into());
    if is_wasm32 {
```

Reply:
Also, do we need that for all wasm32 targets? What about emscripten - doesn't it define its own stdio and such?